cabal-hie/src/CabalHie/Main.hs

```diff
@@ -109,15 +131,11 @@ generateHie tracer opts = do
 
         -- next we collect (directory, selector) pairs
         let componentPaths :: (Semigroup a, CL.HasBuildInfo a) => String -> C.CondTree C.ConfVar [d] a -> Peu r [(Path Absolute, Text)]
-            componentPaths selector comp0  = do
-                (unit, _cn, _ci) <- maybe (die tracer $ "Cannot find unit for " ++ selector) return $
-                    Map.lookup (T.pack selector) components
-
-                let (_, comp) = simplifyCondTree ghcInfo (Map.mapKeys toCabal $ Plan.uFlags unit) comp0
-
-                return $ map (\dir -> (absDir dir, T.pack selector)) (L.toListOf (CL.hsSourceDirs . traverse) comp)
-
-        -- libraries
+            componentPaths selector comp0  =
+                maybe (putWarning tracer WCan'tFindUnit ("Cannot find unit for " ++ selector) $> [])
```

Comment:
Oh, I would prefered a `case`. This `maybe is unreadable. Sorry for not being explicit. I can do that change myself too later.